packages/react-sdk/src/components/image.tsx

```diff
@@ -1,24 +1,129 @@
 import React, { forwardRef, type ElementRef, type ComponentProps } from "react";
+import warnOnce from "warn-once";
+import {
+  generateImgAttrs,
+  allSizes,
+  type ImageLoader,
+} from "../component-utils/image";
+
+/**
+ * Default image loader in case of no loader provided
+ * https://developers.cloudflare.com/images/image-resizing/url-format/
+ **/
+const cloudflareImageLoader: ImageLoader = ({ width, src, quality }) => {
+  if (process.env.NODE_ENV !== "production") {
+    warnOnce(
+      !allSizes.includes(width),
+      "Width must be only from allowed values"
+    );
+  }
+
+  const options = `width=${width},quality=${quality},format=auto`;
+  // Cloudflare docs say that we don't need to urlencode the path params
+  const pathname = `/cdn-cgi/image/${options}/${src}`;
+  return pathname;
+};
+
+const getInt = (x: unknown): number | undefined => {
+  if (typeof x === "number") {
+    return Math.round(x);
+  }
+
+  if (typeof x === "string") {
+    const v = Number.parseInt(x, 10);
+    if (`${v}` === x) {
+      return v;
+    }
+  }
+
+  return undefined;
+};
+
+const DEFAULT_QUALITY = 80;
+
+/**
+ * DEFAULT_SIZES Just an assumption that most images (except hero and icons) are 100% wide on mobile and 50% on desktop.
+ * For icons width are usually set explicitly so DEFAULT_SIZES is not applied.
+ * For hero images, we can allow in UI to select sizes=100vw explicitly.
+ * Anyway, the best would be to calculate this based on canvas data from different breakpoints.
+ * See ../component-utils/image for detailed description
+ **/
+const DEFAULT_SIZES = "(min-width: 1280px) 50vw, 100vw";
 
 const defaultTag = "img";
 
-type ImageProps = ComponentProps<typeof defaultTag>;
+type ImageProps = ComponentProps<typeof defaultTag> & {
+  quality?: number;
+  loader?: ImageLoader;
+  optimize?: boolean;
+};
 
 export const Image = forwardRef<ElementRef<typeof defaultTag>, ImageProps>(
-  (props, ref) => <img {...props} ref={ref} />
+  (
+    {
+      quality: qualityProp,
+      loader: loaderProp,
+      optimize: optimizeProp,
+      width: widthProp,
+      height: heightProp,
+      ...imageProps
+    },
+    ref
+  ) => {
+    const width = getInt(widthProp);
+    const height = getInt(heightProp);
+
+    const quality = Math.max(
+      Math.min(getInt(qualityProp) ?? DEFAULT_QUALITY, 100),
+      0
+    );
+
+    const sizes =
+      imageProps.sizes ?? (width == null ? DEFAULT_SIZES : undefined);
+
+    const loader = loaderProp ?? cloudflareImageLoader;
+
+    // To support previous image behaviour, disable optimize by default
+    const optimize = optimizeProp ?? false;
+
+    const imageAttributes =
+      imageProps.src != null && imageProps.src != ""
+        ? optimize
+          ? generateImgAttrs({
+              src: imageProps.src,
+              width,
+              quality,
+              sizes,
+              loader,
+            })
+          : { src: imageProps.src }
+        : { src: imagePlaceholderSvg };
```

Comment:
I would try to move all of it into component utils, add a fewtests and this way allow someone to create a component in a different library than react while keeping the entire logic as-is